csvy/readers.py

```diff
@@ -96,11 +98,11 @@ def read_metadata(
 
 
 def read_to_array(
-    filename: Union[Path, str],
+    filename: Path | str,
```

Comment:
In theory, this syntax is not valid for Python 3.9. See https://www.slingacademy.com/article/union-type-in-python-the-complete-guide/#:~:text=Introduced%20in%20Python%203.10%20as%20part%20of%20PEP

I think this might be running the `ruff` pre-commit with the wrong python version. Could you add a default python version in the `pre-commit-congig.yml` file? See https://[pre-commit](https://pre-commit.com/#top_level-default_language_version).com/#top_level-default_language_version

Reply:
So in readers.py you've got `from __future__ import annotations`, so the `|` operator should be valid. Not sure why it's not complaining in writers.py though!

That said, when I run `ruff check --target-version py39` it doesn't complain - weird!

Reply:
I forgot about the `__futures__` stuff. Yeah, with that all of that should work. But I agree it should fail if not imported.